hluti1.sql

```diff
@@ -0,0 +1,174 @@
+--Liður 1
+-- Using a FULL OUTER JOIN to fetch all kingdoms and houses, including those without matches
+SELECT
+    k.name AS kingdom_id,
+    h.name AS house_id
+FROM
+    atlas.kingdoms k
+FULL OUTER JOIN
+    got.houses h ON k.name = h.region;
+
+-- Create a new table if it does not exist, with a unique constraint to prevent duplicate mappings
+CREATE TABLE IF NOT EXISTS baratheon.tables_mapping (
+    kingdom_name TEXT,
+    house_name TEXT,
+    UNIQUE(kingdom_name, house_name)
+);
```

Comment:
En þið eruð ekki með þessa dálka, þið eruð með kingdom_id, house_id, location_id, og síðustu 2 eru unique hvor um sig (ekki parið).

```suggestion
CREATE TABLE tables_mapping (
    house_id    INTEGER UNIQUE REFERENCES got.houses(id),
    kingdom_id  INTEGER REFERENCES atlas.kingdoms(gid),
    location_id INTEGER UNIQUE REFERENCES atlas.locations(gid)
);
```